src/gc/bits.d

```diff
@@ -85,6 +85,68 @@ struct GCBits
         return core.bitop.btr(data, i);
     }
 
+    // return non-zero if bit already set
+    size_t setLocked(size_t i) nothrow
+    {
+        version(Windows)
+        {
+            version (D_InlineAsm_X86)
+            {
+                asm @nogc nothrow {
+                    naked; // assume RAX=this, [esp+4]=i
+                    mov ECX, data[EAX];
+                    mov EDX,[ESP+4];
+                    lock;
+                    bts dword ptr[ECX], EDX;
+                    sbb EAX,EAX;
+                    ret 4;
+                }
+            }
+            else version (D_InlineAsm_X86_64)
+            {
+                asm @nogc nothrow {
+                    naked; // assume RCX=this, RDX=i
+                    mov RAX, data[RCX];
+                    lock;
+                    bts qword ptr[RAX], RDX;
+                    sbb RAX,RAX;
+                    ret;
```

Comment:
Is it possible to not use inline asm here?

I can see that both inline and non-inline branches are semantically the same, however I'm surprised by the difference in complexity.

I would have expected instead:
```
atomicFence();
return bts(data, i);
```

Reply:
> I would have expected that using mixed atomicLoad / bts / atomicFence also be doable.

No, a fence is not good enough. it must be an atomic operation, antoher thread might read/write the same word concurrently.

Reply:
> Is it possible to not use inline asm here?

I'd love to use an intrinsic for `lock bts`. Does gdc have it?

Reply:
I can only think of `__atomic_fetch_or`, but that's a `lock cmpxchg`.

Reply:
`__atomic_fetch_or` could work, too. It's basically the same loop I implemented for the generic version.

Reply:
I managed to coerce a `lock bts` out of gcc.  All I needed was a `cast(size_t)`. :-)

https://explore.dgnu.org/z/G-rB79

Reply:
Pretty amazing. LDC has a similar intrinsic, but doesn't seem to be able to optimize it to `lock bts`. I'll add both nonetheless.

Reply:
Could the static assert be removed still anyway?  No point blocking out compilers that don't implement dmd-style asm.

Why isn't the asm good enough for the posix targets?

Reply:
> Could the static assert be removed still anyway?

You mean for GDC on Windows? I've added a version for the GDC builtin now.

> Why isn't the asm good enough for the posix targets?

Calling convention. I didn't want to go through all the possible architectures for naked assembly. But it's probably ok to just use a regular function. The profiler shows the atomic operation is the culprit, anything else in that function is almost never sampled.